src/ui/actions/network.ts

```diff
@@ -123,11 +123,20 @@ export function selectAndFetchRequest(requestId: RequestId): UIThunkAction {
       payload: { frames: formattedFrames, point: timeStampedPoint.point },
     });
 
+    /*
+    These API calls don't directly return anything. Instead:
+
+    - on app setup, `webconsole/actions/network.ts` does a search for network requests and their sub-pieces, and also subscribes to future network requests
+    - in this thunk, we conditionally call `getResponse/RequestBody()`
+    - that API call _by itself_ doesn't return anything...
+    - but when the server sends back data, it triggers those event handlers/callbacks to add the data to Redux
+  */
+
     if (requestSummary.hasResponseBody) {
-      ThreadFront.fetchResponseBody(requestId);
+      replayClient.getResponseBody(requestId);
     }
     if (requestSummary.hasRequestBody) {
-      ThreadFront.fetchRequestBody(requestId);
+      replayClient.getRequestBody(requestId);
     }
```

Comment:
This is our change to re-think these awkward/confusing patterns. (Or defer re-thinking them to later, but let's not copy them over as-is.)

Reply:
I'm not sure how much of this can be "re-thought" atm.

The backend behavior is what really defines the calls here, as the bodies come back via events instead of direct responses.

I agree I don't _like_ this pattern at all. But we can't change how the protocol API returns these values, so the logic for triggering the requests and handling the events is going to be essentially the same regardless of where the event handling lives in the code.

Reply:
> I agree I don't like this pattern at all. But we can't change how the protocol API returns these values

I don't think we should try to change the protocol at this point. I agree that's a big lift.

> so the logic for triggering the requests and handling the events is going to be essentially the same regardless of where the event handling lives in the code.

If we move this functionality into `ReplayClient` – the methods should do something meaningful (like collect and return the values). If there happens to be other listeners attached (in the legacy code) that also observes this information, that's okay.

I feel really strongly that we shouldn't copy over non-obvious patterns like this to the new code though.

Reply:
Honestly I don't think it's worth us removing this stuff from `ThreadFront` at all if we're just copying it as-is into `ReplayClient`. I'd rather just leave it in TF>

Reply:
But literally moving code verbatim from one place, to another, doesn't buy us anything.